### PR TITLE
feat(Alert): add Alert component

### LINKED ISSUE #1082: [Alert] Component for Inline toast-like system messages
## Short Description

Info, warn, and error boxes used to display "inline" system messages. For example, highlighting an error or update within a `ContentCard`. The `Alert` component is **not** modal, and lives in the document flow.

### Mockup

cc @hillarypollak 

Screenshot 2023-11-13 at 2 32 15 PM

### Responsibilities

**This component should be responsible for the following:**
- Dismiss callback
- Content props
- Allowing arbitrary JSX inside the content area
- `isOpen` prop. If the component controls this, we can ensure the Alert is in an aria-live region

**This component should NOT be responsible for...**
- Timing (no auto dismiss)
- Dismissal (handled by consumer via `onUserDismiss` and `isOpen` prop)

### Props Signature

TK

```
<Alert
  propA={}
  propB={}
```

### PR DESCRIPTION
**Pull request by akdetrick**
closes #1082 

Adds `Alert` component for rendering system messages within document flow. Has an `aria-live` region built in.

Screenshot 2023-11-13 at 6 28 59 PM

### CODE REVIEW

File: src/Alert/index.jsx
Comment (akdetrick) [inline src/Alert/index.jsx:24]: This is the live region. When you use an `Alert`, this div will _always_ get rendered. This is useful because assistive technology will announce the alert that renders inside it when `isActive` changes to `true`.

**Review by francesbhughes — APPROVED** [review]
Looks good!

**Comment by akdetrick** [comment]
Awaiting design approval before merging

**Pull request merged**
